https://github.com/flatpak/xdg-desktop-portal/pull/1100

From b734c44545276141b785dd31ff9c0c78a7801baa Mon Sep 17 00:00:00 2001
From: "Anna (navi) Figueiredo Gomes" <navi@vlhl.dev>
Date: Fri, 8 Sep 2023 22:07:06 +0200
Subject: [PATCH 1/3] meson.build: allow linux to build without flatpak
 installed

the old automake scripts would allow for building without flatpak, as the
xml file would just not be copied (because it doesn't exist) so it
wouldn't be used to generate files.

this allows similar behaviour by not requiring the flatpak dependency in
meson.

Signed-off-by: Anna (navi) Figueiredo Gomes <navi@vlhl.dev>
---
 meson.build | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 2254c0d..2366d9e 100644
--- a/meson.build
+++ b/meson.build
@@ -21,8 +21,7 @@ endif
 
 flatpak_intf_dir = get_option('flatpak-interfaces-dir')
 if flatpak_intf_dir == ''
-    flatpak_required = host_machine.system() in ['linux']
-    flatpak_dep = dependency('flatpak', version: '>= 1.5.0', required : flatpak_required)
+    flatpak_dep = dependency('flatpak', version: '>= 1.5.0', required : false)
     if flatpak_dep.found()
       flatpak_intf_dir = flatpak_dep.get_variable(pkgconfig: 'interfaces_dir')
     endif
-- 
2.42.0

